core/modules/utils/auth.js

```diff
@@ -0,0 +1,172 @@
+/*\
+title: $:/core/modules/utils/auth.js
+type: application/javascript
+module-type: utils-node
+
+Utility functions related to hash based authentication.
+
+\*/
+(function(){
+
+/*jslint node: true, browser: false */
+/*global $tw: false */
+"use strict";
+if($tw.node) {
+	var crypto = require("crypto"),
+			fs = require("fs"),
+			path = require("path");;
+}
+
+//	iv + '$' + cipher('aes-256-gcm', randomBytes + date , secret) + '$' + tag
```

Comment:
It would be helpful to have some references to authoritative sources so that others can verify the logic here (I'm assuming this is standard crypto stuff but I'm no expert).